cmd/operator/main.go

```diff
@@ -265,11 +265,20 @@ func Main() int {
 		return 1
 	}
 
-	pao, err := prometheusagentcontroller.New(ctx, cfg, log.With(logger, "component", "prometheusagentoperator"), r)
-	if err != nil {
-		fmt.Fprint(os.Stderr, "instantiating prometheus-agent controller failed: ", err)
-		cancel()
-		return 1
+	var pao *prometheusagentcontroller.Operator
+	ok, err := prometheusagentcontroller.Runnable(ctx, cfg, log.With(logger, "component", "prometheusagentoperator"))
```

Comment:
On second thought, it might be more readable to have the core logic in main() + the pkg/k8sutil package instead of mixing between here and pkg/prometheus/agent.

Reply:
Move `Runnable()` to main?

Reply:
I believe so since AFAIS apart from the `verbs` there is nothing specific to prometheus agent so most of the code should be able to be extracted and here we could maybe just have a mapping between the CR and the necessary verbs and call `Runnable` with that.

```
	verbs := map[string][]string{
		monitoringv1alpha1.PrometheusAgentName:                           {"get", "list", "watch"},
		fmt.Sprintf("%s/status", monitoringv1alpha1.PrometheusAgentName): {"update"},
	}
        ok, err := k8sutil.Runnable(ctx, cfg, log.With(logger, "component", "prometheusagentoperator"), "PrometheusAgent", verbs)

Reply:
basically what Joao said.